### DISCUSSION TITLE
2024-07-29 월요일

### DISCUSSION THREAD

**Discussion by Captain-Kim** [nbc-final-a10/TripBuddy#122] [기록보드]
2024-07-29 월요일 기록보드입니다.

## 여행가고십조(A10) 일일 공통 일과표

| 시간 | 일과 |  기록 |
| :-: | :-: | :-: |
| 09:00 ~ 10:00 | 아침 회의 | |
| 10:00 ~ 11:00 | 기술면접 준비 | |
| 11:00 ~ 13:00 | 집중 개발 time | |
| 13:00 ~ 14:00 | 점심 식사 | |
| 14:00 ~ 18:00 | 집중 개발 time | |
| 18:00 ~ 19:00 | 저녁 식사 | |
| 19:00 ~ 20:00 | 저녁 회의 | <img src="https://github.com/user-attachments/assets/a6602055-fc0f-441b-9342-9efc85de275f" width="300" height="300"/> |
| 20:00 ~ 21:00 | 5분 기록보드 작성 / 개인 회고 | |

```
## 개인 기록보드 서식 (김병준)

### 오늘 한 일

- 내용

### 트러블 슈팅

- 내용

### 내일 계획

- 내용
```

```
// 이미지 테이블
<table>
  <tr>
    <th>이미지 1번 제목</th>
    <th>이미지 2번 제목</th>
  </tr>
  <tr>
    <td><img src="이미지 주소" width="300" height="300"/></td>
    <td><img src="이미지 주소" width="300" height="300"/></td>
  </tr>
</table>
```

**Comment by Captain-Kim** [comment]
## 개인 기록보드 서식 (김병준)

### 오늘 한 일

- 글쓰기 페이지 기능 구현
- 마이페이지 UI 구현
- 검색 페이지와의 공유 컴포넌트 확실하게 재사용 가능하도록 로직 수정 (각종 옵션 props로 제공하도록)

### 트러블 슈팅

- 나무위키의 이미지는 막혀있다. 자세히는 모르겠지만 크롤링을 막아둔 듯 하다. next.config.mjs에서 호스트 서버를 설정해도 호출이 안 된다. 은님과 함께 vercel 에러를 살펴 보니, "프론트 엔드 서버와 작성한 코드에는 문제가 없으니 호스트에게 문의하라"는 문구가 출력되었다.
  - 나무위키가 중요한 게 아니라, 외부 이미지를 끌어다 사용할 때 도대체 왜 안 되겠는지 모르겠는 경우 이 케이스를 확인해보면 좋을 듯.
- Image 컴포넌트를 사용하는 경우에 rounded-full을 사용해서 정원으로 사진을 만들었고 width, height의 속성을 주어 이미지의 크기를 직접 정하였다면 ---->>> className에도 같은 크기를 지정해주어야 원이 안 찌그러짐.
- 동적 라우팅을 생성하고 url path에서 params를 추출할 때 Next.js의 useParams 훅을 사용할 것인가, 직접 추출해서 하위 컴포넌트에 Props로 전달할 것인가?
  - -> 어차피 공수는 둘 다 비슷하다고 판단함.
  - -> useParams 훅을 사용한다고 해서 페이지 컴포넌트에서 훅을 호출하고, 하위 컴포넌트에 자동으로 값이 전달되는 게 아니라 하위 컴포넌트에서도 각각 훅을 호출해주어야 함.
  - -> Props를 전달하는 방식은 하위 컴포넌트에서 Props를 각각 받아주어야 함. 심지어 Props의 타입 지정까지 해주어야 함. -> 물론 이것은 type 폴더 내에서 모듈로 작성하여 하위 컴포넌트에서는 import만 함. 추후 타입을 수정해야 할 때 모든 컴포넌트에 들어가서 수정하는 것보다 이 편이 훨씬 유지보수에 유리할 것이라고 판단함.
  - -> 버디 프로필 페이지 컴포넌트에는 현재 기준 4개의 하위 컴포넌트를 호출하고 있음. 현재는 모두 Params를 필요로 하지만 나중에 Params 필요 없이 단순 렌더링만 하는 컴포넌트가 생길 지 장담을 못함. -> 그렇게 되면 그 컴포넌트에는 useParams 훅을 호출 했는지 안 했는지 컴포넌트 파일을 열어봐야만 알 수 있기에 페이지 컴포넌트에서 Props를 전달하는 게 훨씬 더 한 눈에 잘 보인다고 판단함.

### 내일 계획

- 내용

**Comment by eunohhh** [comment]
## 개인 기록보드 서식 (오은)

### 오늘 한 일

- 온보딩 페이지 UI 구현
- 네이버 소셜로그인 완성

### 트러블 슈팅

- 온보딩 페이지 funnel 하나 넘어갈때마다 fetch를 할 것인가... 마지막에 몰아서 할 것인가...그것이 문제로다

### 내일 계획

- 온보딩 페이지 / buddies api route handler 완성
- 온보딩 지역선택 2뎁스로
- 버디즈 라우트 핸들러 손보기

**Comment by hyunjeongkwak** [comment]
**오늘 한 일**
- 검색 페이지 구성 요소 수정(multi range slider 구현 못함...)

**내일 할 일**
- 검색 기능 개발 들어가기 <- 데이터 들어간 글 목록이 필요
- 자동 스크롤 구현
- ~~검색 페이지 아코디언 삭제 UI 수정...!!~~
- 탑 버튼 추가하기

**Comment by slsl2** [comment]
### 오늘 한 일
- 채팅 메시지 supabase 연결

### 내일 할 일
- 채팅 기능 초안 완성
- 디자인까지 반영해서
- 리얼타임으로
- 채팅 목록 페이지 supabase 연결 (할 수 있는만큼...)
